src/main/resources/lang/de_de.json

```diff
@@ -0,0 +1,121 @@
+{
+	"language": "Deutsch",
+
+	"mapping_format.enigma_file": "Enigma‐Datei",
+	"mapping_format.enigma_directory": "Enigma‐Verzeichnis",
+	"mapping_format.enigma_zip": "Enigma‐ZIP‐Archiv",
+	"mapping_format.tiny_v2": "Tiny‐v2‐Datei",
+	"mapping_format.tiny_file": "Tiny‐Datei",
+	"mapping_format.srg_file": "SRG‐Datei",
+	"mapping_format.proguard": "ProGuard‐Datei",
+	"type.methods": "Methoden",
+	"type.fields": "Felder",
+	"type.parameters": "Parameter",
+	"type.classes": "Klassen",
+
+	"menu.file": "Datei",
+	"menu.file.jar.open": "Java‐Archiv öffnen …",
```

Comment:
Why is there a space before the '…'? (same for all the ones below)

Reply:
When asked whether ellipsis is written with or without spaces, it is important to note what exactly the ellipsis stands for:

- When one or more letters of a word are replaced by ellipsis, they are added to the adjacent word without spaces.
- When entire words or parts of sentences are replaced by ellipsis, they are surrounded by spaces, just as the omitted words would have been.

Reply:
That's not what the ellipsis stands here for, though. It's there because it opens a dialog, not because it omits a part of the sentence.

Reply:
Oh, one of the answers even says:

> Notice that every item in a menu that first opens a dialog requiring additional information must be labelled with a trailing ellipsis (...) (e.g. Save As..., Open...). **There's no space between the menu item and the "...".** A simple confirmation dialog is not considered a dialog that requires additional information.

Reply:
I looked up some German sources for this (as it might be different across languages), but could not find a source mentioning dialogs in particular.
Tbh, I don't see a reason to add ellipsis here at all, as they do not replace anything. It would make sense for e.g. "Save as …", as the ellipsis likely stand for the file name.

Reply:
The English version says "export...", not "save as...". According to the Microsoft language portal (dictionary of technical terms), "export..." translates to "exportieren..."
![image](https://user-images.githubusercontent.com/13084089/79693008-2f48e880-8260-11ea-8824-2b9a6ab5b93b.png)

Reply:
It's not different. Would send a screenshot of a Windows system application but I don't have a Windows installation here right now, but I can guarantee it's the same there. That this is even an argument is mindboggling to me, since you can literally confirm it shouldn't have any space before the '…' by looking at any other application's menus.

![image](https://user-images.githubusercontent.com/3987560/79692903-12b0ae80-8268-11ea-8737-347bba210e57.png)

Reply:
> When entire words or parts of sentences are replaced by ellipsis, they are surrounded by spaces, just as the omitted words would have been.

I have never heard this before in my life. An ellipsis is punctuation, and like most other punctuation, there is no space between it and the preceding word.

Reply:
It's a thing, in both [English][1] ([sometimes?][3]) and [German][2], yeah. But that's not what's happening here.

[1]: https://www.thepunctuationguide.com/ellipses.html
[2]: https://www.annika-lamer.de/beliebte-rechtschreibfehler-das-leerzeichen/
[3]: https://en.wikipedia.org/wiki/Ellipsis#In_British_English

Reply:
The interpretation of the rules I mentioned above is certainly quite controversial.

As @2xsaiko noted, Microsoft generally does not put spaces in front of ellipses in German.
Apple, however, does so: https://youtu.be/BcTj_t0VLp8?t=82

Reply:
The current official German ruleset does not seem to mention this rule explicitly, even though the examples do use it: https://grammis.ids-mannheim.de/rechtschreibung/6210#
The wikipedia article does mention the rule explicitly, but none of the sources seem to.

Reply:
Huh, interesting. Apparently some applications on Linux also put spaces before ellipses. Goes to show that I've only really used German localization back when I also used Windows, I guess.

I still think that the space should be left away, though.

Reply:
I think the space before the ellipsis should definitely be included; it's the same in de_DE Minecraft translation rules